### ISSUE TITLE
Custom component to wrap `flux:select multiple` doesn't render `flux:option`s properly

### ISSUE THREAD

**Issue by jeffchown** [livewire/flux#678]
Hi all,

@calebporzio This might not be an issue but instead my not understanding the order in which Blade interpolation occurs (or something along those lines).

In order to make my code more concise when using the new `flux:select multiple` functionality, I have created a custom component that 'wraps' `flux:select`.  My component is as follows:

resources/views/components/multiselect.blade.php
```
<flux:select
    variant="listbox"
    multiple
    indicator="checkbox"
    {{ $attributes }}
>
    {{ $slot }}
</flux:select>
```

So I can call it in my views via `<x-multiselect ... />`

When I use my new component, the `flux:option`s do not render properly (obvious because the styling doesn't match a normal `flux:option`), nor are they selectable.

For example, if in my view I have:

```
<div class="space-y-4">
    <flux:select variant="listbox" multiple placeholder="Choose industries..." label="via flux:select">
        <flux:option>Photography</flux:option>
        <flux:option>Design services</flux:option>
        <flux:option>Web development</flux:option>
        <flux:option>Accounting</flux:option>
        <flux:option>Legal services</flux:option>
        <flux:option>Consulting</flux:option>
        <flux:option>Other</flux:option>
    </flux:select>

    <x-multiselect placeholder="Choose industries..." label="via x-multiselect">
        <flux:option>Photography</flux:option>
        <flux:option>Design services</flux:option>
        <flux:option>Web development</flux:option>
        <flux:option>Accounting</flux:option>
        <flux:option>Legal services</flux:option>
        <flux:option>Consulting</flux:option>
        <flux:option>Other</flux:option>
    </x-multiselect>
</div>
```

For the first 'native' `flux:select`, the options render and work properly to show:
![Image](https://github.com/user-attachments/assets/1accd0e8-ca99-4968-bcd0-438be4a844de)

but the 2nd select, using my `x-multiselect` component does not render the options properly, and they are not selectable:
![Image](https://github.com/user-attachments/assets/5af59fba-2100-4f91-97a9-98ff1ae25f0a)

Inspecting the results in devtools shows multiple differences in the code generated by the examples above - most notably, no `ui-option`s are in the markup for the 2nd select.

There may be something I'm missing or do not understand, but I'd really like to figure this one out as it will help me greatly simplify my code for some very complex forms that use many multiple select inputs.

Thanks in advance for any help/insights!

**Comment by jeffchown** [comment]
@ju5t Agreed.  I've wrapped other Flux components without issue.
The `flux:select` variants are definitely more complex, but I'm hoping to find a way to do this.

Thanks!  Here's hoping we can get this working.

**Comment by jeffchown** [comment]
UPDATE: It seems the issue is the `flux:option`s in my custom `x-multiselect` are not picking up the `$variant` via the `@aware([ 'variant', 'indicator' ])` directive in the `option.blade.php` file.

Therefore, the `$variant` prop is null and the `$custom` var is set to null causing the file to render as a 'standard' `option` thus breaking the Flux functionality.

Now the trick is to figure out why the `@aware` isn't aware ;)

**Comment by jeffchown** [comment]
Because of the way `@aware` works, the quickest way to accomplish what I want and 'get to green' for now seems to be to create two wrapping custom components, one for `flux:select` and one for `flux:option`.

views/components/multiselect.blade.php
```
@props([
    'variant' => 'listbox',
    'multiple' => true,
    'indicator' => 'checkbox',
])

<flux:select
    :variant="$variant"
    :multiple="$multiple"
    :indicator="$indicator"
    {{ $attributes }}
>
    {{ $slot }}
</flux:select>
```

views/components/multioption.blade.php
```
@aware(['variant', 'indicator'])

@props([
    'variant' => 'listbox',
    'indicator' => 'checkbox',
])

<flux:option :variant="$variant" :indicator="$indicator" {{ $attributes }}>
    {{ $slot }}
</flux:option>
```

Then, I can use the components in my forms via e.g.:
```
<x-multiselect wire:model="industries" placeholder="Choose more industries...">
    <x-multioption>Photography</x-multioption>
    <x-multioption>Design services</x-multioption>
    <x-multioption>Web development</x-multioption>
    <x-multioption>Accounting</x-multioption>
    <x-multioption>Legal services</x-multioption>
    <x-multioption>Consulting</x-multioption>
    <x-multioption>Other</x-multioption>
</x-multiselect>
```

While this works, I'm open to feedback/ideas :)

**Comment by calebporzio** [comment]
Yeah, you're running into a kinda deep nuance of `@aware` in Blade components.

You can achieve what you want by directly passing `variant="listbox"` to each option like so:

```
<x-multiselect>
    <flux:option variant="listbox">...</flux:option>
</x-multiselect>
```

OR you can pass it into `x-multiselect` like so:

```
<x-multiselect variant="listbox">
    <flux:option>One</flux:option>
    <flux:option>Two</flux:option>
    <flux:option>Three</flux:option>
</x-multiselect>
```

If one of those aren't your cup of tea, I'm afraid your only other option is, like you said, wrapping `x-multiselect`.

I understand this is kind of frustrating. This was really the only way I could think of to make the API I wanted work. If it's an issue for others maybe we can consider another API for selects.

**Comment by jeffchown** [comment]
Thanks, @calebporzio 

I learned as much once I dug into how `@aware` works (learned something new!).

When code-diving into Flux's select, I realized I could pass the `variant="listbox"` each time to avoid the need for `x-multioption`.

That said, my main goal in creating the `x-multiselect` (and, later, `x-multioption`) components was to leverage `flux:select` while standardizing the multiselect selects in my apps' codebases and provide a single point of maintenance should I wish to change all of the multiselects in an app (e.g. I had previously been using SlimSelect and was *very* happy to replace it with `flux:select`)

I wasn't really frustrated, just didn't initially know why things weren't working as expected until I found the `@aware` in `flux:option`.  Now that I found it, and understand its 'behaviour', I'll know to look for it next time.

**I understand the decision to use it within Flux to support composability and wouldn't change that - I find your design choices in Flux have made its API and usage very intuitive and flexible.**

A possible solution would be a section in the docs re: customization to mention the usage of `@aware` in the Flux codebase, the possible obstacles it can create when wrapping Flux components in our own custom components (or just a link to https://laravel.com/docs/11.x/blade#accessing-parent-data) and a list of the Flux components that use it - so we can be aware/prepared.
